### PR TITLE
render external link info, even when external file is missing

### LINKED ISSUE #18: h5tree: report external group links in the right place
in h5structure.h5structure._renderGroup()
The problem is the link file and path need to be fed into the next call to _renderGroup().  No such design exists now for that.

### LINKED ISSUE #149: BUG h5tree reports the name of value as defined in the external file
At this time, tree reports the name of (external link) value as defined in the external file. This is wrong. It should be the name as defined in the master file.  In this example, the master file has link to the NXdata group in the external file.

### external data file

```text
/tmp/external.h5
  external_data_group:NXdata
    @NX_class = NXdata
    @signal = x
    x:float64 = 0.123
```

### master data file

```text
/tmp/master.h5 : NeXus data file
  entry:NXentry
    @NX_class = NXentry
    external_data_group:NXdata  <--- should be  "master_data_group:NXdata"
      @NX_class = NXdata
      @signal = x
      x:NX_FLOAT64 = 0.123
```

### PR DESCRIPTION
**Pull request by prjemian**
- FIX #18
- FIX #149

### CODE REVIEW

**Comment by prjemian** [comment]
All tests pass.  Ready for review.

**Comment by prjemian** [comment]
@carterbox Nice!

**Review by carterbox — APPROVED** [review]
Tests seem reasonable. I don't have a firm grasp of the codebase yet, so most of my feedback is documentation/API related.

File: punx/h5tree.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def report(self, show_attributes=True):
             tree_string_list = self._renderGroup(f, txt, indentation="")
         return tree_string_list
 
-    def _renderGroup(self, obj, name, indentation="  "):
+    def _renderGroup(self, obj, name, indentation="  ", md=None):
         """return a [formatted_string] with the contents of the group"""
```
[the remark this answers is not in the document]
Comment (prjemian) [reply punx/h5tree.py]: Will do.

File: punx/utils.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def isHdf5Link(obj):
     if not hasattr(obj, "parent"):
         return False
     details = obj.parent.get(obj.name, getlink=True)
-    return isinstance(details, h5py.HardLink)
+    return isinstance(details, h5py.HardLink)  # TODO: h5py.SoftLink, too?
```
[the remark this answers is not in the document]
Comment (prjemian) [reply punx/utils.py:81]: @carterbox Thanks for the review and suggestions.  I'll follow up on this shortly.

[the remark this answers is not in the document]
Comment (prjemian) [reply punx/utils.py:81]: These are handled slightly different.  Only recently has NeXus decided to accept soft links.

[the remark this answers is not in the document]
Comment (prjemian) [reply punx/utils.py:81]: But this function reports if an item is an HDF5 link (of any sort).

**Comment by prjemian** [comment]
@carterbox Thanks!

**Pull request merged**